src/Mod/TechDraw/Gui/QGIViewBalloon.cpp

```diff
@@ -608,16 +629,24 @@ void QGIViewBalloon::draw()
     }
 
     auto vp = static_cast<ViewProviderBalloon*>(getViewProvider(getViewObject()));
-    if ( vp == nullptr ) {
+    if (!vp) {
         return;
     }
 
     m_lineWidth = Rez::guiX(vp->LineWidth.getValue());
 
     double textWidth = balloonLabel->getDimText()->boundingRect().width();
     double textHeight = balloonLabel->getDimText()->boundingRect().height();
-    float x = Rez::guiX(balloon->X.getValue() * refObj->getScale());
-    float y = Rez::guiX(balloon->Y.getValue() * refObj->getScale());
+    float x, y;
+    // when not dragging take the X/Y properties otherwise the current label position
+    if (!dragged) {
+        x = Rez::guiX(balloon->X.getValue() * refObj->getScale());
+        y = Rez::guiX(balloon->Y.getValue() * refObj->getScale());
+    }
+    else {
+        x = Rez::guiX(balloonLabel->X() / refObj->getScale());
+        y = -Rez::guiX(balloonLabel->Y() / refObj->getScale());
```

Comment:
The label's `X()` and `Y()` are already GUI-side, so shouldn't this just be:
```suggestion
        x = balloonLabel->X();
        y = -balloonLabel->Y();
```

Reply:
Indeed, thanks. I changed this now.